crowbar_framework/lib/repositories.rb

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Repositories
+  class Check
+    def initialize(options = {})
+      @platform = options.fetch :platform
+      @name = options.fetch :name
+    end
+
+    def validate
+      ret = {}
+      ret[:dir_available] = validate_repo_dir
+      ret[:key_md5] = validate_key_file
+      ret[:required] = requirement
+      ret[:web_url] = web_url
+      ret
+    end
+
+    protected
+
+    #
+    # config
+    #
+    def load_config
```

Comment:
To me this method name sounds like it will load the config into the object rather than return it (I would prefer `get_config` or `read_config), although I can't really argue with it since it's already consistent with `YAML.load_file` ;-)